#!/bin/sh /usr/share/dpatch/dpatch-run
## 09-initscript.dpatch by Michael Meskes <meskes@debian.org>
##
## DP: Some changes to initscript.

@DPATCH@

diff -Naurp virtualbox-ose-2.1.0-dfsg.orig/src/VBox/Installer/linux/vboxdrv.sh.in virtualbox-ose-2.1.0-dfsg/src/VBox/Installer/linux/vboxdrv.sh.in
--- virtualbox-ose-2.1.0-dfsg.orig/src/VBox/Installer/linux/vboxdrv.sh.in	2008-12-18 16:30:08.000000000 +0100
+++ virtualbox-ose-2.1.0-dfsg/src/VBox/Installer/linux/vboxdrv.sh.in		2008-12-18 16:36:51.000000000 +0100
@@ -32,7 +32,8 @@
 LOG="/var/log/vbox-install.log"
 NOLSB=%NOLSB%
 
-[ -f /lib/lsb/init-functions ] || NOLSB=yes
+# Include virtualbox-ose defaults if available
+[ -f /etc/default/virtualbox-ose ] && . /etc/default/virtualbox-ose
 [ -f /etc/vbox/vbox.cfg ] && . /etc/vbox/vbox.cfg
 
 if [ -n "$INSTALL_DIR" ]; then
@@ -150,8 +151,10 @@
         if ! rm -f $DEVICE; then
             failure "Cannot remove $DEVICE"
         fi
-        if ! modprobe vboxdrv > /dev/null 2>&1; then
-            failure "modprobe vboxdrv failed. Please use 'dmesg' to find out why"
+	if [ "$LOAD_VBOXDRV_MODULE" = 1 ]; then
+	        if ! modprobe vboxdrv > /dev/null 2>&1; then
+        	    failure "modprobe vboxdrv failed. Please use 'dmesg' to find out why"
+		fi
         fi
         sleep .2
     fi
@@ -181,8 +184,10 @@
         rmmod vboxdrv 2>/dev/null
         failure "Cannot change owner $GROUPNAME for device $DEVICE"
     fi
-    if ! modprobe vboxnetflt > /dev/null 2>&1; then
-        failure "modprobe vboxnetflt failed. Please use 'dmesg' to find out why"
+    if [ "$LOAD_VBOXDRV_MODULE" = 1 ]; then
+	if ! modprobe vboxnetflt > /dev/null 2>&1; then
+        	failure "modprobe vboxnetflt failed. Please use 'dmesg' to find out why"
+	fi
     fi
     succ_msg
 }
